### PR TITLE
build: update golang version to 1.20

### PR DESCRIPTION
**Pull request by 0Tech**
## Description

closes: #1024

## Checklist:

- [x] I followed the [contributing guidelines](https://github.com/Finschia/finschia-sdk/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/Finschia/finschia-sdk/blob/main/CODE_OF_CONDUCT.md).
- [x] I have added a relevant changelog to `CHANGELOG.md`
- [ ] I have added tests to cover my changes.
- [x] I have updated the documentation accordingly.
- [ ] I have updated API documentation `client/docs/swagger-ui/swagger.yaml`

### CODE REVIEW

**Comment by 0Tech** [comment]
I MUST revert the replace on ostracon.

File: crypto/keyring/keyring.go [hunk 1]
```diff
@@ -755,7 +755,7 @@ func newRealPrompt(dir string, buf io.Reader) func(string) (string, error) {
 				continue
 			}
 
-			if err := os.WriteFile(dir+"/keyhash", passwordHash, 0o555); err != nil {
+			if err := os.WriteFile(dir+"/keyhash", passwordHash, 0o600); err != nil {
```
Comment (zemyblue) [inline crypto/keyring/keyring.go:758]: Why does this change?

File: snapshots/manager_test.go [hunk 2]
```diff
@@ -79,7 +79,7 @@ func TestManager_Take(t *testing.T) {
 		Height: 5,
 		Format: snapshotter.SnapshotFormat(),
 		Chunks: 1,
-		Hash:   []uint8{0xcd, 0x17, 0x9e, 0x7f, 0x28, 0xb6, 0x82, 0x90, 0xc7, 0x25, 0xf3, 0x42, 0xac, 0x65, 0x73, 0x50, 0xaa, 0xa0, 0x10, 0x5c, 0x40, 0x8c, 0xd5, 0x1, 0xed, 0x82, 0xb5, 0xca, 0x8b, 0xe0, 0x83, 0xa2},
+		Hash:   []uint8{0x14, 0x38, 0x97, 0x96, 0xba, 0xe4, 0x81, 0xaf, 0x6c, 0xac, 0xff, 0xa5, 0xb8, 0x7e, 0x63, 0x4b, 0xac, 0x69, 0x3f, 0x38, 0x90, 0x5c, 0x7d, 0x57, 0xb3, 0xf, 0x69, 0x73, 0xb3, 0xa0, 0xe0, 0xad},
```
Comment (zemyblue) [inline snapshots/manager_test.go:82]: Why is this snapshot hash changing?

File: store/rootmulti/snapshot_test.go [hunk 3]
```diff
@@ -128,7 +128,7 @@ func TestMultistoreSnapshot_Checksum(t *testing.T) {
 			"aa048b4ee0f484965d7b3b06822cf0772cdcaad02f3b1b9055e69f2cb365ef3c",
 			"7921eaa3ed4921341e504d9308a9877986a879fe216a099c86e8db66fcba4c63",
 			"a4a864e6c02c9fca5837ec80dc84f650b25276ed7e4820cf7516ced9f9901b86",
-			"ca2879ac6e7205d257440131ba7e72bef784cd61642e32b847729e543c1928b9",
+			"8ca5b957e36fa13e704c31494649b2a74305148d70d70f0f26dee066b615c1d0",
```
Comment (zemyblue) [inline store/rootmulti/snapshot_test.go:131]: Why only this change?

File: crypto/keyring/keyring.go (hunk 1 above)
Comment (0Tech) [reply crypto/keyring/keyring.go:758]: The linter complains. Please refer: https://github.com/Finschia/finschia-sdk/pull/1043/commits/c4ffcdcaf9b7639efe4030bf182f412776a1544a

File: snapshots/manager_test.go (hunk 2 above)
Comment (0Tech) [reply snapshots/manager_test.go:82]: Please refer: #1025

File: store/rootmulti/snapshot_test.go (hunk 3 above)
Comment (0Tech) [reply store/rootmulti/snapshot_test.go:131]: Please refer: #1025

**Comment by 0Tech** [comment]
Or, we should merge #1025 into main and deal with the differences between 1.19 and 1.20.

**Comment by zemyblue** [comment]
> Or, we should merge #1025 into main and deal with the differences between 1.19 and 1.20.
> 
> What do you think @zemyblue ?

No I think it's not a good idea.

**Review by zemyblue — APPROVED** [review]

**Review by jaeseung-bae — APPROVED** [review]

**Pull request merged**